packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -289,6 +300,10 @@ abstract class BaseFlutterTask extends DefaultTask {
     Boolean previewDart2
     File sourceDir
     File intermediateDir
+    @Optional
```

Comment:
I believe you have to specify `@Input` as well on `previewDart2`, `extraFrontEndOptions`, and `extraGenSnapshotOptions`. Otherwise Gradle will not re-execute the task, if one of these changes and nothing else does.